CMakeLists.txt

```diff
@@ -251,15 +251,18 @@ if(DEACTIVATE_AVX2)
 endif()
 
 # flags
-# @TODO: set -Wall
+# Set -Wall and other useful warning flags.
+if(CMAKE_C_COMPILER_ID STREQUAL GNU OR CMAKE_C_COMPILER_ID STREQUAL Clang OR CMAKE_C_COMPILER_ID STREQUAL Intel)
+    add_compile_options(-Wall -Wwrite-strings -Wno-unused-function)
+endif(CMAKE_C_COMPILER_ID STREQUAL GNU OR CMAKE_C_COMPILER_ID STREQUAL Clang OR CMAKE_C_COMPILER_ID STREQUAL Intel)
```

Comment:
Repeating the condition in the `endif(..)` is no longer necessary I think, so this could be just `endif()`.

Reply:
Repeating the condition in the `endif(...)` was just a way to see at a glance where the condition block finished.  This is more useful in nested conditions, and we could remove it from here (not a big deal though).

Reply:
@FrancescAlted Aha, I thought it was a left-over from < 2.5 times, when it was mandatory. Yes, no big deal, and could just as well be kept.